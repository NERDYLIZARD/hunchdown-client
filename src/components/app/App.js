/* eslint-disable import/no-named-as-default */
import React from 'react';
import PropTypes from 'prop-types';
import { Route, Switch } from 'react-router-dom';
import NavigationBar from '../common/NavigationBar';
import HomePage from '../pages/HomePage';
import HunchPage from '../boxes/BoxDetail';
import NotFoundPage from '../pages/NotFoundPage';
import BoxPage from '../boxes/BoxPage';
import Modal from '../common/Modal';
import BoxSelectionForm from "../editor/BoxSelectionForm";

// This is a class-based component because the current
// version of hot reloading won't hot reload a stateless
// component at the top-level.
class App extends React.Component
{
  render () {
    return (
      <div>
        <NavigationBar/>
        <Switch>
          <Route exact path="/" component={HomePage}/>
          <Route path="/boxes/:id" component={HunchPage}/>
          <Route path="/boxes" component={BoxPage}/>
          <Route path="/test" component={BoxSelectionForm}/>
          <Route component={NotFoundPage}/>
        </Switch>
        <Modal/>
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.element
};

export default App;
