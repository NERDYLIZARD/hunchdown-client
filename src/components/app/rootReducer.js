import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import cards from '../card/cardReducer';
import activeCard from '../card/activeCardReducer';

const rootReducer = combineReducers({
  cards,
  activeCard,
  routing: routerReducer,
});

export default rootReducer;
