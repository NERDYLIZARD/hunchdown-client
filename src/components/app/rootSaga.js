/**
 * Created on 14-May-18.
 */
import { all } from 'redux-saga/effects';
import { cardSagaWatchers }  from '../card/cardSagas';

export default function* rootSaga() {
  yield all ([
    ...cardSagaWatchers,
  ]);
}
