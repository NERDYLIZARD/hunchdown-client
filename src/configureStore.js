import { applyMiddleware, compose, createStore } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import createSagaMiddleware from 'redux-saga'
import createHistory from 'history/createBrowserHistory';
// 'routerMiddleware': the new way of storing route changes with redux middleware since rrV4.
import { routerMiddleware } from 'react-router-redux';
import reducers from './reducers';
import saga from './sagas';
import apiMiddleware from './middlewares/api';
import editorModalMiddleware from './middlewares/editor-modal';

export const history = createHistory();
const sagaMiddleware = createSagaMiddleware();

function configureStoreProd(initialState) {
  const reactRouterMiddleware = routerMiddleware(history);
  const middlewares = [
    // Add other middleware on this line...

    reactRouterMiddleware,
    sagaMiddleware,
    apiMiddleware,
    editorModalMiddleware
  ];

  const store = createStore(reducers, initialState, compose(
    applyMiddleware(...middlewares)
    )
  );

  sagaMiddleware.run(saga);

  return store;
}

function configureStoreDev(initialState) {
  const reactRouterMiddleware = routerMiddleware(history);
  const middlewares = [
    // Add other middleware on this line...

    // Redux middleware that spits an error on you when you try to mutate your state either inside a dispatch or between dispatches.
    reduxImmutableStateInvariant(),
    reactRouterMiddleware,
    sagaMiddleware,
    apiMiddleware,
    editorModalMiddleware
  ];

  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools
  const store = createStore(reducers, initialState, composeEnhancers(
    applyMiddleware(...middlewares)
    )
  );

  sagaMiddleware.run(saga);

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./reducers', () => {
      const nextReducer = require('./reducers').default; // eslint-disable-line global-require
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}

const configureStore = process.env.NODE_ENV === 'production' ? configureStoreProd : configureStoreDev;

export default configureStore;
