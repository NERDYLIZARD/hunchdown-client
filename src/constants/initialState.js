export default {
  cards: {},  // associative array
  card: {
    wisdom: '',
    attribute: '',
  }
};
