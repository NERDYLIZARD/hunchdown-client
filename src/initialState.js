export default {
  hunches: {
    byId: {}, // associative array
    editing: {
      modalOpen: false,
      hunch: {
        id: '',
        slug: '',
        wisdom: '',
        attribute: '',
      }
    }
  },
  boxes: {
    byId: {}, // associative array
    editing: {
      modalOpen: false,
      box: {
        id: '',
        title: '',
        description: '',
      }
    }
  },
};
