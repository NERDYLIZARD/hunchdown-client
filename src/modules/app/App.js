/* eslint-disable import/no-named-as-default */
import React from 'react';
import PropTypes from 'prop-types';
import { Route, Switch } from 'react-router-dom';
import NavigationBar from './NavigationBar'
import HomePage from '../pages/HomePage';
import CardPage from '../cards/components/CardPage';
import NotFoundPage from '../pages/NotFoundPage';

// This is a class-based component because the current
// version of hot reloading won't hot reload a stateless
// component at the top-level.

class App extends React.Component {
  render() {
    return (
      <div>
        <NavigationBar />
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route path="/cards" component={CardPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.element
};

export default App;
