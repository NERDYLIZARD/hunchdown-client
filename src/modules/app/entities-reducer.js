/**
 * Created on 15-Aug-18.
 */
import merge from 'lodash/merge';

export const initialState = {
  boxes: {},
  hunches: {},
};

export default function entitiesReducer(state = initialState, action) {
  if (action.payload && action.payload.entities) {
    return merge({}, state, action.payload.entities);
  }
  return state;
}
