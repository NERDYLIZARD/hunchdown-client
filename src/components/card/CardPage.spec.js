/**
 * Created on 07-Apr-18.
 */

import React from 'react';
import { shallow } from 'enzyme';
import ConnectedCardPage, { CardPage } from './CardPage';
import initialState from '../../constants/initialState';
import configureMockStore from 'redux-mock-store';
import { ConnectedRouter } from 'react-router-redux';
import {history} from "../app/configureStore";
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';


describe("<CardPage />", () => {
  const actions = {
    loadCards: jest.fn(),
  };

  function setup() {
    const props = {
      cards: initialState.cards,
      actions: actions,
    };
    return shallow(<CardPage {...props} />)
  }

  it('should contain h1 with "Cards" as its value', () => {
    const wrapper = setup();
    expect(wrapper.find('h1').text()).toBe('Cards');
  });

  it('should call loadCards()', () => {
    setup();
    expect(actions.loadCards).toBeCalled();
  });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)(initialState);

    const component = create(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <ConnectedCardPage/>
        </ConnectedRouter>
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});

