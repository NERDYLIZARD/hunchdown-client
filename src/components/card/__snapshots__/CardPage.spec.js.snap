// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardPage /> should match snapshot 1`] = `
<div>
  <h1>
    Cards
  </h1>
  <a
    href="/cards/new"
    onClick={[Function]}
  >
    New Card
  </a>
</div>
`;
